Makefile

```diff
@@ -58,7 +58,7 @@ test: unit ## Run all the tests
 # Run the unit tests
 .PHONY: unit
 unit: ## Run the unit tests
-	go test $(GO_TEST_FLAGS) -coverprofile $(COVER_PROFILE) ./...
+	go test -race $(GO_TEST_FLAGS) -coverprofile $(COVER_PROFILE) ./...
```

Comment:
I am still wondering if we should also add e.g `-count 20`, which should help detect possible concurrency problems faster (if I am not mistaken), but the tests will also take longer to run.